src/rgw/rgw_admin_opt_role.h

```diff
@@ -0,0 +1,38 @@
+#ifndef CEPH_RGW_ADMIN_OPT_ROLE_H
+#define CEPH_RGW_ADMIN_OPT_ROLE_H
+
+
+#include "rgw_rest_conn.h"
+#include "rgw_realm_watcher.h"
+
+#include "common/ceph_json.h"
+#include "rgw_role.h"
+#include "rgw_admin_common.h"
+
+int handle_opt_role_get(const string& role_name, const string& tenant, CephContext *context,
```

Comment:
I suggest have a base class for admin command than one for each admin command type (role, user, multisite ...)
.you can look how we do it for Rest API see rgw_rest.h rgw_rest_s3.h and rgw_rest_admin.h
I think something like command(enum opt) with method parse, execute , get_error